src/Plugin.Maui.Audio/AudioPlayer/AudioStopwatch.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Diagnostics;
+
+namespace Plugin.Maui.Audio;
+
+public class AudioStopwatch(TimeSpan startOffset, double speed) : Stopwatch
```

Comment:
What is the need for this class? Why can't we rely on the information reported back from the platforms players?

Reply:
because there is a bug in the AVAudioPlayer and it doesn't report the correct time. There is a bug that I have linked in the code and I don't think that will ever be resolved.

Reply:
To reproduce the bug, play the previous version on android for a while, maybe change the speed and play around with the settings. Sometimes, the Position of the audio will jump!

Reply:
Ok thanks. Sorry I've tried reviewing on my phone to help get this through but I've made a mess at points. I'll try and take a look at the rest when I get my laptop out over the weekend. Unless @jfversluis is happy with it

Reply:
Don't worry 😊 I'd be glad if you go through it and run your own tests and am happy to explain why I did it what I did.

Reply:
When you say there is a bug in `AVAudioPlayer` do you mean `MediaPlayer` on android instead? I can only see that `AudioStopwatch` is used on Android.

Reply:
I've only been able to test on iOS and macOS but it all seems happy :)

Reply:
Oh yes apologies, I meant the MediaPlayer on Android